targets/TARGET_STM/can_api.c

```diff
@@ -118,7 +179,8 @@ static void _can_init_freq_direct(can_t *obj, const can_pinmap_t *pinmap, int hz
     Phase_segment_2            | 30 tq          | <nts2> = <n_tq> - 1 - <nts1>
     Synchronization_Jump_width | 30 tq          | <nsjw> = <nts2>
     */
-    int ntq = 10000000 / hz;
+
+    int ntq = _can_get_core_frequency() / hz;
```

Comment:
Maybe yo could try HAL_RCCEx_GetPeriphCLKFreq with RCC_PERIPHCLK_FDCAN ?

Reply:
Mmm it seems that STM32H7 doesn't provide the same function... :-(

Reply:
Note you can try code like this:
````
#ifdef TARGET_STM32G4
int ntq = HAL_RCCEx_GetPeriphCLKFreq(RCC_PERIPHCLK_FDCAN)/ hz;
#else
// STM32H7 doesn't support yet HAL_RCCEx_GetPeriphCLKFreq for FDCAN
// Internal ST ticket 92465
int ntq = 10000000 / hz;
#endif

````

Reply:
This is a really nice catch, thank you! I added your proposal in.